CRM/Core/Permission/Drupal8.php

```diff
@@ -92,4 +92,33 @@ public function upgradePermissions($permissions) {
     }
   }
 
+  /**
+   * Given a roles array, check user has one of those roles
+   *
+   * @param array $array
+   *   The roles to check.
+   *
+   * @return bool
+   *   true if yes, else false
+   */
+  public function checkGroupRole($array) {
+    if (isset($array)) {
+
+      // This returns a 0-indexed array of role machine names, e.g.
+      // [
+      //   0 => 'authenticated',
+      //   1 => 'administrator',
+      // ]
+      $user_roles = \Drupal::currentUser()->getRoles();
```

Comment:
@demeritcowboy any thoughts to using array_flip and using array_key_exists instead in L116?

Reply:
Yeah I thought about improving it but just left it as-is from DrupalBase. Yeah might be worth updating it while it's in my head. Need to head out soon - will update later.

Reply:
Updated.